### DISCUSSION TITLE
User Authentication

### DISCUSSION THREAD

**Discussion by MucahitErdoganUnlu** [bounswe/bounswe2024group11#253] [Q&A]
Our current solution for authentication on Django uses TokenAuthentication. However, we may want to change it because:
* If you use TokenAuthentication in production you must ensure that your API is only available over https.
* For an implementation which allows more than one token per user, has some tighter security implementation details, and supports token expiry, please see the [Django REST Knox](https://github.com/James1345/django-rest-knox) third party package.

Which one should be our implementation?

**Comment by umitcan07** [comment]
I think I would stick to TokenAuthentication. Keep it simple & stupid :)

**Comment by MucahitErdoganUnlu** [comment]
A useful website: [DRF Token Authentication vs Third Party Packages (JWT and Django Rest Knox)](https://medium.com/@jayshridotnet/drf-token-authentication-vs-third-party-packages-jwt-and-django-rest-knox-c9e68c541e16)
Especially, this part is important:
* However, in scenarios where users frequently log in from multiple devices, and device independence is crucial, alternative authentication solutions that support per-client tokens may offer a more user-friendly experience.

Since our project supports login from multiple device types, let alone multiple devices, using Third Party Packages is a better alternative.
